NitroxPatcher/Patches/Dynamic/EnergMixin_SpawnDefaultAsync_Patch.cs

```diff
@@ -0,0 +1,45 @@
+using System.Collections.Generic;
+using System.Reflection;
+using System.Reflection.Emit;
+using HarmonyLib;
+using NitroxModel.Helper;
+using UnityEngine;
+
+namespace NitroxPatcher.Patches.Dynamic;
+
+/// <summary>
+/// When items are spawned in Subnautica will automatically add batteries to them async.  This is challenging to deal with because it is 
+/// difficult to discriminate between these async/default additions and a user purposfully placing batteries into an object. Instead, we
+/// disable the default behavior and allow Nitrox to always spawn the batteries.  This guarentees we can capture the ids correctly.
+/// </summary>
+public class EnergyMixin_SpawnDefaultAsync_Patch : NitroxPatch, IDynamicPatch
+{
+    public static readonly MethodInfo TARGET_METHOD_ORIGINAL = Reflect.Method((EnergyMixin t) => t.SpawnDefaultAsync(default(float), default(TaskResult<bool>)));
+    public static readonly MethodInfo TARGET_METHOD = AccessTools.EnumeratorMoveNext(TARGET_METHOD_ORIGINAL);
+
+    public static readonly OpCode INJECTION_OPCODE = OpCodes.Call;
+    public static readonly object INJECTION_OPERAND = Reflect.Method(() => CrafterLogic.NotifyCraftEnd(default(GameObject), default(TechType)));
+
+    public static IEnumerable<CodeInstruction> Transpiler(MethodBase original, IEnumerable<CodeInstruction> instructions)
+    {
+        Validate.NotNull(INJECTION_OPERAND);
+
+        // Blank out the method and replace with:
```

Comment:
```suggestion
        // Blanks out the generated MoveNext() and replaces it with:
```
Minor: To better reflect that we are blanking/replacing the generated state machine.